source/mesh_refinement/isosurfaces.cc

```diff
@@ -0,0 +1,401 @@
+            }
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+  }
+}
+
+// explicit instantiations
+namespace aspect
+{
+  namespace MeshRefinement
+  {
+    ASPECT_REGISTER_MESH_REFINEMENT_CRITERION(Isosurfaces,
+                                              "isosurfaces",
+                                              "A mesh refinement criterion that computes "
+                                              "refinement indicators between two iso-surfaces of "
+                                              "specific field entries (e.g. temperature, composition)."
+                                              "\n\n"
+                                              "The way these indicators are derived on each isosurface is by "
+                                              "checking whether the solutions of specific "
+                                              "fields are within the ranges of values given. If these conditions "
+                                              "hold, then an indicator is either put on or taken off "
+                                              "in order to secure mesh refinement within the range of levels "
+                                              "given. Usage of this plugin allows the user to put a conditional "
+                                              "minimum and maximum refinement function onto fields that they "
+                                              "are interested in."
+                                              "\n\n"
+                                              "For now, only temperature and compositional fields are allowed as "
+                                              "field entries. The key words could be 'Temperature' or one of the names "
+                                              "of the compositional fields which are either specified by user or set up "
+                                              "as C\\_0, C\\_1, etc."
+                                              "\n\n"
+                                              "Usage: A list of isosurface separated by semi-colons (;). Each "
+                                              "isosurface entry consists of multiple entries separated by a comma. "
+                                              "The first two entries indicate the minimum and maximum refinement "
+                                              "levels respectively. The entries after the first two describe the "
+                                              "fields the isosurface applies to, followed by a colon (:), which again "
+                                              "followed by the minimum and maximum grid levels separated by bar (|). An "
+                                              "example for an isosurface is '0, 2, Temperature: 300 | 600; 2, 2, C\\_1: 0.5 | 1'. "
+                                              "If isosurfaces overlap, the last one sets the values. "
```

Comment:
```suggestion
                                              "If isosurfaces overlap, the last isosurface entry specified in the input file dictates refinement. "
```

Reply:
It would also be good to add how this plugin interacts with other refinement strategies. It will override strategies like viscosity or temperature, but how does it interact with `minimum refinement function` for example? Does the order in which the tag_additional_cells strategies are evaluated matter then?